File: lib/business/Thresholds.dart
class Thresholds {
  static final String verybad = 'Rất tệ';
  static final String bad = 'Tệ';
  static final String normal = 'Bình thường';
  static final String medium = 'Trung bình';
  static final String good = 'Tuyệt';
  static final String verygood = 'Rất tuyệt';

  static final String safe = 'An toàn';
  static final String extreme = 'Cực nguy hiểm';

  static final String verycool = 'Rất lạnh';
  static final String cool = 'Lạnh';
  static final String hot = 'Nóng';
  static final String veryhot = 'Rất nóng';

  static final String low = "Thấp";
  static final String verylow = "Rất thấp";
  static final String high = "Cao";
  static final String veryhight = 'Rất cao';

  static String aqi(double value) {
    String message = '';

    if (value >= 0 && value < 51) message = Thresholds.verygood;
    if (value >= 51 && value < 101) message = Thresholds.normal;
    if (value >= 101 && value < 151) message = Thresholds.medium;
    if (value >= 151 && value < 201) message = Thresholds.bad;
    if (value >= 201 && value < 301) message = Thresholds.verybad;
    if (value >= 301) message = Thresholds.extreme;

    return message;
  }

  static String uv(double value) {
    String message = '';

    if (value >= 0 && value < 2) message = Thresholds.safe;
    if (value >= 2 && value < 4) message = Thresholds.medium;
    if (value >= 4 && value < 7) message = Thresholds.high;
    if (value >= 7 && value < 10) message = Thresholds.veryhight;
    if (value >= 10) message = Thresholds.extreme;

    return message;
  }

  static String temp(double value) {
    String message = '';

    if (value < 0) message = Thresholds.extreme;
    if (value >= 0 && value < 10) message = Thresholds.verylow;
    if (value >= 10 && value < 20) message = Thresholds.low;
    if (value >= 20 && value < 25) message = Thresholds.normal;
    if (value >= 25 && value < 35) message = Thresholds.high;
    if (value >= 35 && value < 40) message = Thresholds.veryhight;
    if (value >= 40) message = Thresholds.extreme;

    return message;
  }

  static String co(double value) {
    String message = '';

    if (value >= 0 && value < 3) message = Thresholds.safe;
    if (value >= 3 && value < 7) message = Thresholds.normal;
    if (value >= 7 && value < 15) message = Thresholds.medium;
    if (value >= 15 && value < 30) message = Thresholds.high;
    if (value >= 30 && value < 62) message = Thresholds.veryhight;
    if (value >= 62) message = Thresholds.extreme;

    return message;
  }
}

File: android/app/src/main/kotlin/com/example/o2_mobile/Thresholds.kt
class Thresholds {
    companion object {
        const val verybad = "Rất tệ"
        const val bad = "Tệ"
        const val normal = "Bình thường"
        const val medium = "Trung bình"
        const val good = "Tuyệt"
        const val verygood = "Rất tuyệt"

        const val safe = "An toàn"
        const val extreme = "Cực nguy hiểm"

        const val verycool = "Rất lạnh"
        const val cool = "Lạnh"
        const val hot = "Nóng"
        const val veryhot = "Rất nóng"

        const val low = "Thấp"
        const val verylow = "Rất thấp"
        const val high = "Cao"
        const val veryhight = "Rất cao"

        fun aqi(value: Double): String {
            var message = ""

            if (value >= 0 && value < 51) message = verygood
            if (value >= 51 && value < 101) message = normal
            if (value >= 101 && value < 151) message = medium
            if (value >= 151 && value < 201) message = bad
            if (value >= 201 && value < 301) message = verybad
            if (value >= 301) message = extreme

            return message
        }

        fun uv(value: Double): String {
            var message = ""

            if (value >= 0 && value < 2) message = safe
            if (value >= 2 && value < 4) message = medium
            if (value >= 4 && value < 7) message = high
            if (value >= 7 && value < 10) message = veryhight
            if (value >= 10) message = extreme

            return message
        }

        fun temp(value: Double): String {
            var message = ""

            if (value < 0) message = extreme
            if (value >= 0 && value < 10) message = verylow
            if (value >= 10 && value < 20) message = low
            if (value >= 20 && value < 25) message = normal
            if (value >= 25 && value < 35) message = high
            if (value >= 35 && value < 40) message = veryhight
            if (value >= 40) message = extreme

            return message
        }

        fun co(value: Double): String {
            var message = ""

            if (value >= 0 && value < 3) message = safe
            if (value >= 3 && value < 7) message = normal
            if (value >= 7 && value < 15) message = medium
            if (value >= 15 && value < 30) message = high
            if (value >= 30 && value < 62) message = veryhight
            if (value >= 62) message = extreme

            return message
        }
    }
}
